src/main/scala/lasp/hapi/util/Time.scala

```diff
@@ -0,0 +1,54 @@
+package lasp.hapi.util
+
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+import java.time.format.DateTimeFormatterBuilder
+import java.time.format.DateTimeParseException
+import java.time.temporal.ChronoField.MONTH_OF_YEAR
+import java.time.temporal.ChronoField.DAY_OF_MONTH
+import java.time.temporal.ChronoField.HOUR_OF_DAY
+import java.time.temporal.ChronoField.MINUTE_OF_HOUR
+import java.time.temporal.ChronoField.SECOND_OF_MINUTE
+
+import cats.implicits._
+
+object Time {
+
+  /**
+   * Parses a HAPI time string.
+   *
+   * @param str HAPI time string
+   */
+  def parse(str: String): Option[LocalDateTime] =
+    tryParse(str, format) <+> tryParse(str, formatOrd)
+
+  private def tryParse(
+    str: String,
+    fmt: DateTimeFormatter
+  ): Option[LocalDateTime] =
+    try {
+      Option(LocalDateTime.parse(str, fmt))
+    } catch {
+      case _: DateTimeParseException => None
+    }
+
+  /** Format for normal date/time strings. */
+  private val format = new DateTimeFormatterBuilder()
+    .appendPattern("yyyy[-MM[-dd]]['T'HH[:mm[:ss]]]['Z']")
```

Comment:
Does the hapi spec say anything about bits smaller than a second? Is this just for time range requests or also for data time stamps?

Reply:
I've added fractions of seconds.